specification.md

```diff
@@ -382,6 +397,112 @@ a resource containing an array of [retry](#Retry-Definition) definition.
+
+
+<details><summary><strong>Click to view example definition</strong></summary>
+<p>
+
+<table>
+<tr>
+    <th>JSON</th>
+    <th>YAML</th>
+</tr>
+<tr>
+<td valign="top">
+
+```json
+{  
+   "time": "PT2M",
+   "workflowId": "createandsendreport"
+}
+```
+
+</td>
+<td valign="top">
+
+```yaml
+time: PT2M
+workflowId: createandsendreport
+```
+
+</td>
+</tr>
+</table>
+
+</details>
+
+The `time` property defines the total amount of time in the execution window. Once a workflow instance is created,
+and the amount of the defined time is reached, the workflow instance needs to be terminated.
+
+The `interrupt` property defines if the currently running instance should be allowed to finish its current 
+execution flow before it needs to be terminated. If set to `true`, the current instance execution must stop immediately.
+
+The `workflowId` property defines an unique id of a sub-workflow which should be executed before the workflow instance
+is terminated.
+
+`ExecWindow` allows us to solve interesting use cases, especially within the IoT (Internet of Things) domain.
+Let's say that we have a sensor that reads temperature values in our living room every 2 minutes. 
+Our workflow acts upon these events, the first one starting a workflow instance, and every consecutive one 
+being correlated to the same instance (given the correlation id being the room name, namely "livingroom" for example).
+After every hour of collecting and processing this data, we want to build a report for all the so far collected temperatures. 
+
+For this example we could define our events in the workflow as:
+
+```json
+{  
+  ...
+  "events": [
+   {  
+     "name": "TemperatureEvent",
+     "type": "roomtemp",
+     "source": "tempsensor",
+     "correlation": [
+      { 
+        "contextAttributeName": "roomid"
+      } 
+     ]
+   }
+  ],
+  ...
+}
+```
+
+Without `execWindow` it would be pretty hard to solve this business problem with control-flow logic alone.
+With it however it becomes pretty easy. Let's look at our sample workflow definition:
+
+```json
+{  
+  ...
+  "name": "Living Room Temperature Collection Workflow",
+  "execWindow": {
+     "time": "PT1H",
+     "workflowId": "sendLivingRoomTempReport"
+  },
+  "states": [
+    ... control-flow logic for collecting and processing room temp event data...
+  ],
+  "events": [
+    ...
+  ],
+  "functions": [
+    ...
+  ]
+}
+```
+
+Here we define the `execWindow` `time` property to "PT1H" meaning that after 1 hour of our workflow instance 
+collecting living room temperature data, we want to stop collecting data, and build our report by invoking the 
+"sendlivingroomtempreport" sub-workflow, before terminating the workflow instance. After the defined 
+amount of time, the very next living room temperature event will create a new instance of our workflow
+which will then again start collecting these events for 1 hour, build the report and terminate its execution, and so on.
```

Comment:
```suggestion
Here we define the `execWindow` `time` property to "PT1H". This means that after 1 hour of the workflow instance collecting living room temperature data, we want to stop collecting it. Then we build the report by invoking the 
"sendlivingroomtempreport" sub-workflow before terminating the workflow instance. 

After the defined amount of time, the very next living room temperature event will create a new instance of the workflow
which will then again start collecting these events for 1 hour, build the report and terminate its execution, and so on.
```
I've found this paragraph too long and hard to understand the first time I read it. Not sure if I got it right, but I tried to keep the sentences smaller. wdyt?